workflows/sweep_simulate.snake

```diff
@@ -0,0 +1,324 @@
+    # removing duplicates if bleft and left are the same or brighta and right
+    windows_bpoints = np.unique(windows_bpoints)
+    # something wonky happening with float conversion
+    if windows_bpoints[-1] == ts.sequence_length - 1:
+        windows_bpoints[-1] = ts.sequence_length
+    assert (windows_bpoints.shape == (2,)) or windows_bpoints.shape == (4,)
+    taj=ts.Tajimas_D(windows=windows_bpoints,mode="site")
+    focal_taj = taj[len(taj)//2]
+    return focal_taj, taj
+
+# ###############################################################################
+# GENERAL RULES & GLOBALS
+# ###############################################################################
+rule all:
+    input:
+        [output_dir +f"/simulated_data/sweeps/boundary_sims/sim_{seed}_{config['region_size']}.trees"
+        for seed in seed2_array
+        ] + [output_dir + f"/simulated_data/sweeps/neutral/{seed}/sim_{chrom}_{left:.0f}_{right:.0f}{suffix}"
+        for left, right in windows
+        for seed in seed_array
+        for suffix in [".trees", ".stats.npz"] 
+        ] + [output_dir + f"/simulated_data/sweeps/bgs/{annot}/{dfe}/{seed}/sim_{chrom}_{left:.0f}_{right:.0f}{suffix}"
+        for left, right in windows
+        for seed in seed_array
+        for dfe in dfe_list
+        for annot in annotation_list
+        for suffix in [".trees", ".stats.npz"] 
+        ] + [output_dir +  f"/simulated_data/sweeps/sweep/{annot}/{dfe}/{coeff}/{time}/{seed}/sim_{chrom}_{left:.0f}_{right:.0f}{suffix}"
+        for left, right in windows
+        for seed in seed_array
+        for dfe in dfe_list
+        for annot in annotation_list
+        for coeff in sel_coeffs
+        for time in times
+        for suffix in [".trees", ".stats.npz"] 
+        ]
+
+print(windows.shape)
+
+rule boundary_sims:
+    input:
+    output:
+        output_dir + "/simulated_data/sweeps/boundary_sims/sim_{seed}_{region_size}.trees"
+    resources: time=6000, mem_mb=100000
+    run:
+        model = species.get_demographic_model(demo_model["id"])
+        mut_rate = model.mutation_rate 
+        # Use new sample specification, give selected pop and sample size as config arguments
+        samples = demo_model["samples"]
+        dfe = species.get_dfe(dfe_list[0])
+        # reducing number of del muts bc it's too much!
+        dfe.proportions = [1-(0.7*0.25), 0.7*0.25]
+        clen = int(wildcards.region_size)
+        contig = species.get_contig(length=clen)
+        contig.add_dfe(intervals=[[0, clen]], DFE=dfe)# applying DFE over some equally spaced 
+        contig.mutation_rate = mut_rate
+        engine = stdpopsim.get_engine("slim")
+        ts = engine.simulate(
+            model,
+            contig,
+            samples,
+            slim_scaling_factor=config["slim_scaling_factor"],
+            slim_burn_in=config["slim_burn_in"],
+            seed=int(wildcards.seed),
+        )
+        ts.dump(output[0])
+
+rule neutral:
```

Comment:
do we have to do these simulations? can we instead use the simulations that are already being produced by the rest of the pipeline for neutral sims?

Reply:
What do you mean?

Reply:
i mean that we might be able to grab simulations from the `n_t.snake` workflow?